### ISSUE TITLE
Timeline animation example does not have an animation

### ISSUE THREAD

**Issue by YunjieLi** [mapbox/mapbox-gl-js#3597]
I notice the [create a timeline animation example](https://www.mapbox.com/mapbox-gl-js/example/timeline-animation/) does not have an animation. I also wonder if d3 for date time conversion and legend is necessary here since it may scare off people who're not familiar w/ d3.

I would like to revise the example and potentially use data driven styling instead.

**Comment by mollymerp** [comment]
@yunjieli we try not to include external libraries like d3 in gl-js examples, but if you want to add DDS / an actual animation option, go ahead! let me know if you need help getting set up!

**Comment by YunjieLi** [comment]
Sounds good @mollymerp . I mean currently the example uses d3 and I'm thinking of removing it since the use if not critical.

**Comment by YunjieLi** [comment]
I poked around with this example today and have some new thoughts:
- I cut the ticket primarily because of the ambiguous title: it's a manual slider not an automated  playback. After chatting w/ @peterqliu I'm now hesitant to add the playback animation here since such an addition does not demonstrate any new use of GL JS. 
- The use of d3 is valid. Keep. (GL JS source can read geojson from an external url, but I think the d3 call makes sense since the geojson is needed elsewhere in the script too.)
- Rewriting the demo using data driven styling would make the codes much cleaner and focused. However, I find it hard to match up the circle and label layers using DDS, and keeping the original layers would be the better option.
![image](https://cloud.githubusercontent.com/assets/12749701/20911217/6e4eaa66-bb1b-11e6-84d0-7c70bee13d42.png)

 In short, on second thought I think the example should stay what it is, except that I propose to change the title from `Creating a timeline animation` to `Creating a time slider`.

cc @mollymerp any thoughts?

**Comment by mollymerp** [comment]
> I'm now hesitant to add the playback animation here since such an addition does not demonstrate any new use of GL JS.

Do we already have an example with a playback animation?

> I find it hard to match up the circle and label layers using DDS, and keeping the original layers would be the better option.

I don't understand this issue. I think it would be preferable to use DDS bc this is the best practice performance wise. Let me know if you want to talk through this.

**Comment by peterqliu** [comment]
>I find it hard to match up the circle and label layers using DDS, and keeping the original layers would be the better option.

per chat, she means using DDS means lumping all circles into one layer, which doesn't let us control their stack order the way splitting them by severity does.

**Comment by YunjieLi** [comment]
@mollymerp we don't have a playback animation in the examples.

OK definitely prefer DDS if that's the case then.

I may be overly perfectionist about the styling but there's a subtle difference between symbol behavior before and after DDS because of the layer stack order ( ^ thanks @peterqliu) 

Before DDS | After DDS
---- | ----
![image](https://cloud.githubusercontent.com/assets/12749701/20943298/bd70d27a-bbb3-11e6-90ac-68058a56f187.png) | ![image](https://cloud.githubusercontent.com/assets/12749701/20943307/c4f88380-bbb3-11e6-9ac6-7ddc0a646ccb.png)

This actually sets me thinking that if we want to keep the code more focused by removing the popup component and even the label layer.

looping in @tristen who's the original author.

**Comment by YunjieLi** [comment]
This merged
